FastAPI integration Async Redis session dependency override, error during test RuntimeError: <Queue > is bound to a different event loop

**Describe the bug**
I am currently developing a FastAPI app that gets a Redis Session from a Dependency Injection, and uses it to write some data on Redis. 
I need to test the endpoint, therefore I created a pytext fixture which in turn:
1. create a FakeAsyncRedis
2. override the dependency in the endpoint
3. return the same FakeAsyncRedis object

in the test basically I call the endpoint, and I read the key I expect to get set, but I get the following error
`test_redis - RuntimeError: <Queue at 0x111538a10 maxsize=0 tasks=1> is bound to a different event loop`


**To Reproduce**
My fixtures are:

```python
@pytest.fixture(scope="session")
def event_loop():
    policy = asyncio.get_event_loop_policy()
    loop = policy.new_event_loop()
    yield loop
    loop.close()


@pytest.fixture
async def redis(event_loop: asyncio.AbstractEventLoop):
    redis_session = fakeredis.FakeAsyncRedis(decode_responses=True, encoding="utf-8", )

    async def _get_async_redis_session() -> AsyncIterator[fakeredis.FakeAsyncRedis]:
        yield redis_session

    for app in all_fast_apps:
        app.dependency_overrides[get_async_redis_session] = _get_async_redis_session
    return redis_session
```
This is my dependency used to fetch a Redis Connection:
```python
from redis import asyncio as aioredis
async def get_async_redis_session() -> AsyncIterator[Redis]:
    async with aioredis.from_url(redisUrl, **redis_args) as session:
        yield session
    await session.close()
```

and this is the endpoint definition:
```python
async def write_on_redis(
    request: Request,
    redis: Redis = Depends(get_async_redis_session)
) :
  await redis.set(name=key, value=value, ex=expire)
```
Finally the test where I get the error:

```python
def test_write(redis):  # redis is the fixture we defined before

       url = "myapiendpoint"
        response = client.get(url)
       value = await redis.get(key) <-- this throws the error
```
**Expected behavior**
Get the key set before in the endpoint 

**Desktop (please complete the following information):**
 - OS: [e.g. iOS]
 - fakeredis==2.21.1
 - redis==4.6.0


<!-- POLAR PLEDGE BADGE START -->
## Upvote & Fund

- We're using [Polar.sh](https://polar.sh/cunla) so you can upvote and help fund this issue.
- We receive the funding once the issue is completed & confirmed by you.
- Thank you in advance for helping prioritize & fund our backlog.

<a href="https://polar.sh/cunla/fakeredis-py/issues/292">
<picture>
  <source media="(prefers-color-scheme: dark)" srcset="https://polar.sh/api/github/cunla/fakeredis-py/issues/292/pledge.svg?darkmode=1">
  <img alt="Fund with Polar" src="https://polar.sh/api/github/cunla/fakeredis-py/issues/292/pledge.svg">
</picture>
</a>
<!-- POLAR PLEDGE BADGE END -->

@cunla thanks for reading :)
Sure, this is the stack trace
```
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
../../.pyenv/versions/3.11.5/envs/li/lib/python3.11/site-packages/redis/asyncio/client.py:513: in execute_command
    conn = self.connection or await pool.get_connection(command_name, **options)
../../.pyenv/versions/3.11.5/envs/li/lib/python3.11/site-packages/redis/asyncio/connection.py:1381: in get_connection
    if await connection.can_read_destructive():
../../.pyenv/versions/3.11.5/envs/li/lib/python3.11/site-packages/redis/asyncio/connection.py:759: in can_read_destructive
    return await self._parser.can_read_destructive()
../../.pyenv/versions/3.11.5/envs/li/lib/python3.11/site-packages/redis/asyncio/connection.py:250: in can_read_destructive
    return await self._stream.read(1)
../../.pyenv/versions/3.11.5/envs/li/lib/python3.11/site-packages/fakeredis/aioredis.py:83: in read
    return await self._socket.responses.get()  # type:ignore
../../.pyenv/versions/3.11.5/lib/python3.11/asyncio/queues.py:155: in get
    getter = self._get_loop().create_future()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <Queue at 0x1152f8510 maxsize=0 tasks=1>

    def _get_loop(self):
        loop = events._get_running_loop()
    
        if self._loop is None:
            with _global_lock:
                if self._loop is None:
                    self._loop = loop
        if loop is not self._loop:
>           raise RuntimeError(f'{self!r} is bound to a different event loop')
E           RuntimeError: <Queue at 0x1152f8510 maxsize=0 tasks=1> is bound to a different event loop
```
Actually, I have same output with and without the event_loop fixture

--- a/docs/index.md
+++ b/docs/index.md
@@ -183,29 +183,64 @@ django_rq.queues.get_redis_connection = get_fake_connection
 
 See info on [this issue][7]
 
+If you're using FastAPI dependency injection to provide a Redis connection,
+then you can override that dependency for testing.
+
+Your FastAPI application main.py:
+
 ```python
-from fakeredis.aioredis import FakeRedis
-from httpx import AsyncClient
-from .main import app  # this is the fastapi app object
-from .main import RedisBackend  # this is a class with a bunch of methods with the Redis code for the actions I perform
+from typing import Annotated, Any, AsyncIterator
 
+from redis import asyncio as redis
+from fastapi import Depends, FastAPI
 
-@pytest_asyncio.fixture(scope="function")
-async def redis_conn() -> AsyncIterator[Redis[str]]:
-    # Actually I have some code here to return fakeredis or real redis according to a pytest 
-    # command-line option, but that's not really relevant. It's the reason I bother here with
-    # the async context, though. There's no cleanup needed with FakeRedis alone.
-    async with FakeRedis(decode_responses=True, version=(6,)) as redis_conn:
-        await redis_conn.flushdb()
-        yield redis_conn
+app = FastAPI()
 
+async def get_redis() -> AsyncIterator[redis.Redis]:
+    # Code to handle creating a redis connection goes here, for example
+    async with redis.from_url("redis://localhost:6379") as client:  # type: ignore[no-untyped-call]
+        yield client
 
-@pytest_asyncio.fixture(scope="function")
-async def client(redis_conn: Redis[str]) -> AsyncIterator[AsyncClient]:
-    with mock.patch("main.create_backend", return_value=RedisBackend(redis_conn)):
-        app.dependency_overrides[_some_other_injected_dependency] = _some_other_override
-        async with AsyncClient(app=app, base_url="http://testserver") as client:
-            yield client
+@app.get("/")
+async def root(redis_client: Annotated[redis.Redis, Depends(get_redis)]) -> Any:
+    # Code that does something with redis goes here, for example:
+    await redis_client.set("foo", "bar")
+    return {"redis_keys": await redis_client.keys()}
+```
+
+Assuming you use pytest-asyncio, your test file
+(or you can put the fixtures in conftest.py as usual):
+
+```python
+from typing import AsyncIterator
+from unittest import mock
+
+import fakeredis
+import httpx
+import pytest
+import pytest_asyncio
+from redis import asyncio as redis
+
+from main import app, get_redis
+
+@pytest_asyncio.fixture
+async def redis_client() -> AsyncIterator[redis.Redis]:
+    async with fakeredis.FakeAsyncRedis() as client:
+        yield client
+
+@pytest_asyncio.fixture
+async def app_client(redis_client: redis.Redis) -> AsyncIterator[httpx.AsyncClient]:
+    async def get_redis_override() -> redis.Redis:
+        return redis_client
+    transport = httpx.ASGITransport(app=app)  # type: ignore[arg-type] # https://github.com/encode/httpx/issues/3111
+    async with httpx.AsyncClient(transport=transport, base_url="http://test") as app_client:
+        with mock.patch.dict(app.dependency_overrides, {get_redis: get_redis_override}):
+            yield app_client
+
+@pytest.mark.asyncio
+async def test_app(app_client: httpx.AsyncClient) -> None:
+    response = await app_client.get("/")
+    assert response.json()["redis_keys"] == ["foo"]
 ```
 
 ## Known Limitations

